flake8_trio.py

```diff
@@ -278,3 +407,5 @@ def run(self) -> Generator[Tuple[int, int, str, Type[Any]], None, None]:
 TRIO100 = "TRIO100: {} context contains no checkpoints, add `await trio.sleep(0)`"
 TRIO101 = "TRIO101: yield inside a nursery or cancel scope is only safe when implementing a context manager - otherwise, it breaks exception handling"
 TRIO102 = "TRIO102: it's unsafe to await inside `finally:` unless you use a shielded cancel scope with a timeout"
+TRIO103 = "TRIO103: except Cancelled or except BaseException block with a code path that doesn't re-raise the error"
+TRIO104 = "TRIO104: Cancelled and BaseException must be re-raised"
```

Comment:
```suggestion
TRIO104 = "TRIO104: Cancelled (and therefore BaseException) must be re-raised"
```
I think the explanation does help to motivate this check.

--------

This does actually suggest that if you have an earlier `except:` clause for `Cancelled` we could do a more precise check.  However:

- I think that you should always re-raise BaseException *anyway* (e.g. for KeyboardInterrupt), and
- it's rare enough not to be worth the trouble here

Reply:
ooh that's a fun corner case. Don't think it would add too much to check for it though so I could add it

Reply:
Nah, not worth the code complexity IMO.

Reply:
Nah, not worth the complexity (code + mental model) IMO, not when it's bad style anyway.

Reply:
makes sense. Fixed :+1: